core/input/input_event.h

```diff
@@ -508,10 +508,33 @@ class InputEventPanGesture : public InputEventGesture {
 };
 
 class InputEventMIDI : public InputEvent {
+public:
+	enum Message {
+		MESSAGE_NONE = 0,
```

Comment:
These should be referencing the enum in `input_enums` to avoid any copy paste errors IMO

Reply:
How? As in, a comment mentioning them?

Reply:
As in `MESSAGE_NONE = MIDI_MESSAGE_NONE`, I tgink the enum in the class should be secondary, the `input_enums` should be the one in charge as it were, otherwise we have a lot of secondary dependencies, the other input related enums aren't local to the events

Reply:
Ok, so the **MIDIMessage** enum will always exist internally, but the assumption is someday it will no longer be exposed in the global scope. Understood.

Reply:
My reasoning being that on the GDScript side it makes sense to make the MIDI stuff self contained, because there isn't really anything else with it, but on the C++ side we have drivers etc. which don't really engage with the input events in the same way, so going through the event to do things feels unnecessary (also less changes to keep track of)